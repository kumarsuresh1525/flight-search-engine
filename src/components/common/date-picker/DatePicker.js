import React, {useState} from 'react';
import './DatePicker.scss';

const DatePicker = ({onChange}) => {
  const [value, setValue] = useState('');
  const handleDateChange = (event) => {
    setValue(event.target.value);
    onChange(event.target.value);
  }
  return (
    <div>
      <input
        className='date-picker'
        type='date'
        value={value}
        onChange={handleDateChange}
      />
    </div>
  );
}

export default DatePicker;