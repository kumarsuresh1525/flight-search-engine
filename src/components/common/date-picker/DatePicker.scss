.date-picker {
  font-size: 14px;
  height: 30px;
  margin-top: 20px;
  width: -webkit-fill-available;
}