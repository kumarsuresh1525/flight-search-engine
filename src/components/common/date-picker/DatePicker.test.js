import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DatePicker from './DatePicker';

describe('Date-Picker test cases', () => {
  const defaultProps = {
    onChange: jest.fn()
  }
  test('renders date-picker correctly', () => {
    const { asFragment } = render(<DatePicker {...defaultProps}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should handle date change', () => {
    const { container } = render(<DatePicker {...defaultProps}/>);
    const datePicker = container.querySelector('.date-picker');
    fireEvent.change(datePicker, {target: {value: '2020-02-24'}});
    expect(datePicker.value).toBe('2020-02-24');
  });
});