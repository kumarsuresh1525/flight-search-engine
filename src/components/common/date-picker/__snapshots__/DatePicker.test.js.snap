// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date-Picker test cases renders date-picker correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      class="date-picker"
      type="date"
      value=""
    />
  </div>
</DocumentFragment>
`;
